include/dataformats/pacman/PACMANFrame.hpp

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file PACMANFrame.hpp PACMAN bit fields and accessors (ND LAr).
+ *
+ * This is part of the DUNE DAQ , copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef DATAFORMATS_INCLUDE_DATAFORMATS_PACMANFRAME_HPP_
+#define DATAFORMATS_INCLUDE_DATAFORMATS_PACMANFRAME_HPP_
+
+#include <bitset>
+#include <iostream>
+#include <vector>
+#include <stdint.h>
+
+namespace dunedaq {
+namespace dataformats {
+
+using word_t = uint32_t; // NOLINT(build/unsigned)
+
+/**
+ * @brief PACMAN frame
+ */
+class PACMANFrame
+{
+  public:
+  #define WORD_LEN   8  // bytes
+  #define HEADER_LEN 16 // bytes
+  
+  struct PACMANHeader
+  {
+    word_t version : 5;
+    // Can be implemented on request, not currently necessary.
+  };
+  
+  enum msg_type { // message type declarations
+    DATA_MSG = 0x44,
+    REQ_MSG  = 0x3F,
+    REP_MSG  = 0x21
+  };
+
+  enum word_type { // word type declarations
+    DATA_WORD  = 0x44,
+    TRIG_WORD  = 0x54,
+    SYNC_WORD  = 0x53,
+    PING_WORD  = 0x50,
+    WRITE_WORD = 0x57,
+    READ_WORD  = 0x52,
+    TX_WORD    = 0x44,
+    ERR_WORD   = 0x45
+  };
+  enum packet_type { // packet type declarations
+    DATA_PACKET         = 0x0,
+    CONFIG_WRITE_PACKET = 0x2,
+    CONFIG_READ_PACKET  = 0x3
+  };
+
+
+  #define MSG_TYPE_OFFSET  0 // bytes
+  #define MSG_WORDS_OFFSET 6 // bytes
+  #define UNIX_TS_OFFSET   1 // bytes
+  #define WORD_TYPE_OFFSET         0 // bytes
+  #define IO_CHANNEL_OFFSET        1 // bytes
+  #define RECEIPT_TIMESTAMP_OFFSET 4 // bytes
+  #define PACKET_OFFSET            8 // bytes
+
+
+  /* ~~~ Access into message header and message contents ~~~ */
+
+  uint32_t get_msg_bytes(void* msg) {
+    // get total number of bytes in message, including header
+    return HEADER_LEN + WORD_LEN * (*get_msg_words(msg));
+  }
+
+  uint8_t* get_msg_type(void* msg) {
+    // get pointer to message type
+    return (uint8_t*)(((uint8_t*)msg) + MSG_TYPE_OFFSET);
```

Comment:
Better to use C++-style casts
`return static_cast<uint8_t*>(msg) + MSG_TYPE_OFFSET;`